src/Nethermind/Nethermind.Facade/Eth/RpcTransaction/LegacyTransactionForRpc.cs

```diff
@@ -95,7 +95,7 @@ public override Transaction ToTransaction()
         tx.GasLimit = Gas ?? 90_000;
         tx.Value = Value ?? 0;
         tx.Data = Input;
-        tx.GasPrice = GasPrice ?? 20.GWei();
+        tx.GasPrice = GasPrice ?? 0;
```

Comment:
This will also set for derived classes: `AccessListTransactionForRpc`
I expect that is fine?

`EIP1559TransactionForRpc` and below have their own logic

Reply:
Yes, exactly. I can't see that being an issue, at least

Reply:
Changed the PR title so it's a bit more clear we're not only dealing with strictly legacy transactions